crates/astria-sequencer/src/bridge/bridge_sudo_change_action.rs

```diff
@@ -76,27 +72,54 @@ impl ActionHandler for BridgeSudoChangeAction {
             "unauthorized for bridge sudo change action",
         );
 
+        if let Some(sudo_address) = self.new_sudo_address {
+            state.put_bridge_account_sudo_address(self.bridge_address, sudo_address);
+        }
+
+        if let Some(withdrawer_address) = self.new_withdrawer_address {
+            state.put_bridge_account_withdrawer_address(self.bridge_address, withdrawer_address);
+        }
+
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl FeeHandler for BridgeSudoChangeAction {
+    // allow: false positive due to proc macro; fixed with rust/clippy 1.81
+    #[allow(clippy::blocks_in_conditions)]
+    #[instrument(skip_all, err(level = Level::WARN))]
+    async fn calculate_and_pay_fees<S: StateWrite>(&self, mut state: S) -> Result<()> {
+        let tx_context = state
+            .get_transaction_context()
+            .expect("transaction source must be present in state when executing an action");
+        let from = tx_context.address_bytes();
         let fee = state
             .get_bridge_sudo_change_base_fee()
             .await
             .wrap_err("failed to get bridge sudo change fee")?;
+
+        ensure!(
+            state
+                .is_allowed_fee_asset(&self.fee_asset)
+                .await
+                .wrap_err("failed to check allowed fee assets in state")?,
+            "invalid fee asset",
+        );
+
         state
-            .get_and_increase_block_fees(&self.fee_asset, fee, Self::full_name())
-            .await
+            .add_fee_to_block_fees(
+                self.fee_asset.clone(),
+                fee,
+                tx_context.transaction_id,
+                tx_context.source_action_index,
+            )
             .wrap_err("failed to add to block fees")?;
         state
-            .decrease_balance(self.bridge_address, &self.fee_asset, fee)
+            .decrease_balance(from, &self.fee_asset, fee)
```

Comment:
Should bridge sudo change action be deducting the fee from the signer as opposed to the bridge address?